### PR TITLE
Add support for check private vulnerability reporting endpoint

### PR DESCRIPTION
**Pull request by Sairaviteja27**
fixes https://github.com/google/go-github/issues/3152

### CODE REVIEW

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:2407]: This should be private since it isn't exposed anywhere to the end-user.

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:2427]: We don't need to explicitly check for `StatusNotFound` below as far as I can tell from the docs, so this could all be simplified to just:

```suggestion
	return privateReporting.Enabled, resp, err
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @Sairaviteja27 !

File: github/repos.go
Comment (gmlewis) [reply github/repos.go:2427]: The only reason I called this out here is because the official documentation doesn't say that it ever returns a "404 Not Found"... hence I don't think we need it in this particular case.

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:2431]: Note that the zero value of a boolean in Go is `false`. Therefore these five lines can be replaced with:

```suggestion
	return privateReporting.Enabled, resp, err
```

**Review by gmlewis — APPROVED** [review]
Thank you, @Sairaviteja27 !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by Sairaviteja27** [comment]
@gmlewis I think code coverage check is failing. Could you suggest if I should I add tests in other methods?

**Comment by gmlewis** [comment]
> @gmlewis I think code coverage check is failing. Could you suggest if I should I add tests in other methods?

Don't worry about that.  We've had some weirdness with codecov lately and I've made two reports with the problems we've seen and don't have any resolution yet.  Thanks, though.

**Review by be0x74a — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @be0x74a !
Merging.

**Pull request merged**
